controllers/metric_controller.go

```diff
@@ -0,0 +1,267 @@
+				{
+					HonorLabels:     true,
+					Interval:        "30s",
+					Path:            "/metrics",
+					Port:            "https-metrics",
+					Scheme:          "https",
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+					TLSConfig: &monv1.TLSConfig{
+						CAFile: "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt",
+						SafeTLSConfig: monv1.SafeTLSConfig{
+							ServerName: &serverName,
+						},
+					},
+					RelabelConfigs: []monv1.RelabelConfig{
+						{
+							Action:      "replace",
+							Regex:       "(.*)",
+							Replacement: &replacement0,
+							TargetLabel: "instance",
+							SourceLabels: []monv1.LabelName{
+								"__meta_kubernetes_endpoint_address_target_name",
+							},
+						},
+						{ // Include only Windows nodes for this serviceMonitor
+							Action: "keep",
+							Regex:  "windows",
+							SourceLabels: []monv1.LabelName{
+								"__meta_kubernetes_node_label_kubernetes_io_os",
+							},
+						},
+						{ // Change the port from the kubelet port 10250 to 9182
+							Action:      "replace",
+							Regex:       "(.+)(?::\\d+)",
+							Replacement: &replacement1,
+							TargetLabel: "__address__",
+							SourceLabels: []monv1.LabelName{
+								"__address__",
+							},
+						},
+						{ // Update the job label from kubelet to windows-exporter
+							Action:      "replace",
+							Replacement: &replacement2,
+							TargetLabel: "job",
+						},
+					},
+				},
+			},
+			NamespaceSelector: monv1.NamespaceSelector{
+				MatchNames: []string{"kube-system"},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": "kubelet",
+				},
+			},
+		},
+	}
+
+	if err == nil {
+		// check if existing serviceMonitor's contents are as expected, delete it if not
+		if existingSM.Name == expectedSM.Name && existingSM.Namespace == expectedSM.Namespace &&
+			reflect.DeepEqual(existingSM.Spec, expectedSM.Spec) {
+			return nil
+		}
+		err = r.ServiceMonitors(r.watchNamespace).Delete(context.TODO(), metrics.WindowsMetricsResource,
+			metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to delete service monitor %s/%s: %w", r.watchNamespace, metrics.WindowsMetricsResource,
+				err)
+		}
+		r.log.Info("Deleted malformed resource", "serviceMonitor", metrics.WindowsMetricsResource,
+			"namespace", r.watchNamespace)
+	}
+
+	_, err = r.ServiceMonitors(r.watchNamespace).Create(context.TODO(), expectedSM, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating service monitor: %w", err)
+	}
+	return nil
+}
+
+// mapToWatchNamespace fulfills the MapFn type, while always returning a request to the operator watch namespace
+func (r *metricReconciler) mapToWatchNamespace(_ context.Context, _ client.Object) []reconcile.Request {
+	return []reconcile.Request{{
+		NamespacedName: kubeTypes.NamespacedName{Namespace: r.watchNamespace, Name: metrics.WindowsMetricsResource},
+	}}
```

Comment:
Shouldn't this only enqueue a request in the  case that client.Object.Namespace == watchNamespace?
Else any servicemonitor in the cluster changing will cause a reconcile to trigger.
That is assuming we are watching all namespaces and not just the wmco watch namespace

Reply:
yes that makes sense, I missed this.